client/components/common/ViewDocumentInModal.jsx

```diff
@@ -0,0 +1,164 @@
+import lightBaseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
+import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
+import getMuiTheme from 'material-ui/styles/getMuiTheme';
+import IconMenu from 'material-ui/IconMenu';
+import MenuItem from 'material-ui/MenuItem';
+import IconButton from 'material-ui/IconButton';
+import MoreVertIcon from 'material-ui/svg-icons/navigation/more-vert';
+import VisibilityIcon from 'material-ui/svg-icons/action/visibility';
+import Dialog from 'material-ui/Dialog';
+import RaisedButton from 'material-ui/RaisedButton';
+import store from '../../store/configureStore';
+import DocumentViewDialog from '../common/DocumentViewDialog';
+import { deleteDocumentAction, getDocumentsAction } from '../../actions/documents.action';
+import EditDialog from './EditDialog';
+
+
+class ViewDocumentInModal extends React.Component {
+
+  constructor(props) {
+    super(props);
+    const storeState = store.getState();
+    const { isAuthenticated, user } = storeState.authentication;
+    this.state = {
+      openDocumentView: false,
+      openDeleteDialog: false,
+      openSnackbar: false,
+      openEditDialog: false,
+      isAuthenticated,
+      user,
+    };
+    this.openDocumentView = this.openDocumentView.bind(this);
+    this.closeDocumentView = this.closeDocumentView.bind(this);
+    this.toggleDeleteDialog = this.toggleDeleteDialog.bind(this);
+    this.openEditDialog = this.openEditDialog.bind(this);
+    this.closeEditDialog = this.closeEditDialog.bind(this);
+  }
+
+  openDocumentView() {
+    this.setState({ openDocumentView: true });
+  }
+
+  closeDocumentView() {
+    this.setState({ openDocumentView: false });
+  }
+
+  toggleDeleteDialog() {
+    this.setState({ openDeleteDialog: !this.state.openDeleteDialog });
+  }
+
+  openEditDialog() {
+    this.setState({ openEditDialog: true });
+  }
+  closeEditDialog() {
+    this.setState({ openEditDialog: false });
+  }
+
+  deleteDocument(documentID) {
+    this.props.dispatch(deleteDocumentAction(documentID))
+      .then(() => {
+        this.toggleDeleteDialog();
+        this.setState({ openSnackbar: true });
+      });
+      // .catch(() => console.log('delete error'));
+  }
+
+  render() {
+    const { ownerID, access, id } = this.props.documentToView;
+    const { documentOwnerRoleID } = this.props.documentToView.User;
+    const userLinks = (
+      <div>
+        {
+          (this.state.isAuthenticated)
+          ? (ownerID === this.state.user.userID || [1, 2].includes(this.state.user.roleID))
+            ? <div>
+              <MenuItem primaryText="Edit" onTouchTap={() => this.openEditDialog()} />
+              <MenuItem primaryText="Delete" onTouchTap={() => this.toggleDeleteDialog()} />
+            </div>
+            :
+            <MenuItem primaryText="Email" />
+            :
+            <MenuItem primaryText="Email" />
+        }
+      </div>
+    );
+
+    const guestLinks = (
```

Comment:
this can live outside the render method too in a `renderGuestLinks` method, for example, and call with `this.renderGuestLinks()`. Same with  `deleteDocementActions`.